.github/workflows/pr-docker.yaml

```diff
@@ -1,13 +1,14 @@
 name: PR Docker
 on:
-  pull_request:
+  pull_request_target:
```

Comment:
I think this is frowned upon because it would allow someone to open a PR with a change to this file that dumps our secrets

Reply:
Just to clarify, could someone change this workflow to run a (new) script which they check in and dumps secrets?

Reply:
If a PR changes this workflow, it won't run in that PR -- it has to be merged for it to run.

It's the same protection against someone adding a new workflow which uses `pull_request_target` that tries to dump secrets

Reply:
got it, thanks

Reply:
can we document what would be an unsafe thing to do directly in this yaml? I'd like to lower the likelihood of someone later changing this file and doing something unsafe

Reply:
Yea, we should definitely put a comment here on why we're using `pull_request_target` and why it's ok

Reply:
Yea it's not ideal, there are two safeguards:
1. CI runs for PRs from external contributors must be approved by a maintainer
2. `pull_request_target` runs in the context of the target repo, not the merge commit 

1) safeguards against a malicious external user running anything (assuming the reviewer is diligent) and 2) prevents the external user from running malicious code (if the reviewer isn't diligent) unless that malicious code has already been committed to the target repo first.  It's not fool proof but it does make the case better.  More details on how all this works can be found here: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

We could also just assume that anyone making changes to workflows already has write access to the repo and change this to a `pull_request` trigger.  It would just mean that maintainers doing CI infra work can't work from forks which is probably ok.  Let me know what you think.

Reply:
I'm actually just going to go ahead and change this to `pull_request`.